probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from probatus.utils import preprocess_data, preprocess_labels,BaseFitComputePlotClass,get_single_scorer
+from  sklearn.model_selection import cross_val_score
+from  sklearn.pipeline import Pipeline
+from  sklearn.impute import SimpleImputer
+from  sklearn.compose import ColumnTransformer
+from  sklearn.preprocessing import OneHotEncoder
+import matplotlib.pyplot as plt
+import numpy as np 
+import pandas as pd 
+
+class CompareImputationStrategies(BaseFitComputePlotClass):
+    """
+    Comparison of various imputation stragegies that can be used for imputation 
+    of missing values. 
+    The aim of this class is to present the model performance based on imputation
+    strategies and choosen model.
+    For models like XGBoost & LighGBM which have capabilities to handle misisng values by default
+    the model performance with no imputation will be shown as well.
+    The missing values categorical features are filled with `missing` and an missing indicator is
+    added.
+
+    Usage E.g.
+    ```python
+
+    #Import the class
+    from probatus.missing.imputation import CompareImputationStrategies
+    #Create the strategies.
+    strategies = {
+       'Simple Median Imputer' : SimpleImputer(strategy='median',add_indicator=True),
+       'Simple Mean Imputer' : SimpleImputer(strategy='mean',add_indicator=True),
+       'Iterative Imputer'  : IterativeImputer(add_indicator=True,n_nearest_features=5,
+       sample_posterior=True),
+       'KNN' : KNNImputer(n_neighbors=3)
+    #Create a classifier.
+    clf = lgb.LGBMClassifier()
+    #Create the comparision of the imputation strategies.
+    cmp = CompareImputationStrategies(
+        clf=clf,
+        strategies=strategies,
+        cv=5,
+        model_na_support=True)
+
+    cmp.fit_compute(X_missing,y)
+    #Plot the results.
+    cmp.plot()
+
+    <img src="../img/imputation_comparision.png" width="500" />
+   }
+
+    ```
+
+    """
+    def __init__(self,clf,strategies,scoring='roc_auc',cv=None,model_na_support=True,n_jobs=-1,verbose=0,
+                random_state=None):
+        """
+        Initialise the class.
+
+        Args :
+            clf(model object):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategies (dictionary of sklearn.impute objects):
+                Dictionary containing the sklearn.impute objects.
+                e.g.
+                
+                strategies = {'KNN' : KNNImputer(n_neighbors=3),
```

Comment:
please run `mkdocs serve` and check if all the tables in the API page are correctly generated.

Reply:
Done.